AsyncDisplayKit/Private/ASPendingStateController.mm

```diff
@@ -98,6 +76,24 @@ - (void)flushNow
   _flags.pendingFlush = NO;
 }
 
+
+#pragma mark Private Methods
+
+/**
+ This method is assumed to be called with the lock held.
+ */
+- (void)scheduleFlushIfNeeded
+{
+  if (_flags.pendingFlush) {
+    return;
+  }
+
+  _flags.pendingFlush = YES;
+  dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Later, we should consider if there is a more optimal time to run this. I bet you that it would be better to have a runloop observer, strategically times it to occur as close to the transaction commit as possible, specifically after the entire main queue has been drained for that runloop.  This is not essential because any developer relying on this kind of timing is going to experience at least some Visual inconsistency / possibilities of properties applying in different frames, but I think it would be worthwhile to investigate further. The right place to start would probably be with some form of logging facility that allows us to record & number turns of the runloop, and look at the empirical behavior of how these flushes interact with user-triggered inputs, layout passes including a large background remeasure / relayout, etc.